File: Lab1/Lab1/T1.cs
using System.Threading;

namespace Lab1
{
    class T1
    {
        public static void Run(Data data)
        {
            data.print_message(Thread.CurrentThread.Name, "started");

            data.B = data.vector(data.isRandom);                    // 1
            data.E = data.vector(data.isRandom);
            //data.print_vector("vector B: ", data.B);
            //data.print_vector("vector E: ", data.E);

            data.sem1.Release(3);                                   // 2-4
            data.sem2.WaitOne();                                    // 5
            data.sem3.WaitOne();                                    // 6
            data.sem4.WaitOne();                                    // 7

            data.mut1.WaitOne();
            int[] D = data.D;                                       // 8
            int[,] ME = data.ME;                                    // 9
            data.mut1.ReleaseMutex();

            data.operation1(D, ME, 0, data.H);                      // 10
            data.print_message(Thread.CurrentThread.Name, "operation 1 complete");

            data.evh1.WaitOne();                                    // 11
            data.operation2(0, data.H * 2);                         // 12
            data.print_message(Thread.CurrentThread.Name, "operation 2 complete");

            data.evh2.WaitOne();                                    // 13
            data.operation3();                                      // 14
            data.print_message(Thread.CurrentThread.Name, "operation 3 complete");

            int ai = data.operation4(0, data.H);                    // 15
            data.print_message(Thread.CurrentThread.Name, "operation 4 complete");

            lock (data._lock)
            {
                data.operation5(ai);                                // 16
            }
            data.print_message(Thread.CurrentThread.Name, "operation 5 complete");

            data.sem5.Release(3);                                   // 17-19
            data.sem6.WaitOne();                                    // 20
            data.sem7.WaitOne();                                    // 21
            data.sem8.WaitOne();                                    // 22

            data.mut2.WaitOne();
            int a = data.a;                                         // 23
            int x = data.x;                                         // 24
            data.mut2.ReleaseMutex();

            data.operation6(a, x, 0, data.H);                       // 25
            data.print_message(Thread.CurrentThread.Name, "operation 6 complete");
            data.bar1.SignalAndWait();                              // 26

        }
    }
}

File: Lab3/Lab3/src/T1.java
public class T1 extends Thread {
    private final Data data;

    public T1(Data data) {
        this.data = data;
    }

    public void run() {
        data.printMessage("T1", "started");

        data.M1.set_MM(Data.matrix());                              // 1
        data.M1.set_B(Data.vector());
        data.M1.set_MX(Data.matrix());

        data.M2.SignalIn();                                         // 2-4
        data.M2.WaitIn();                                           // 5-6

        int d1 = data.M1.get_d();                                   // 7
        int[] Z1 = data.M1.get_Z();                                 // 8

        data.M1.operation1(d1, Z1, 0, data.H);                      // 9
        data.printMessage("T1", "operation 1 complete");

        data.M3.WaitSort1();                                        // 10

        data.M1.operation2(0, data.H*2);                            // 11
        data.printMessage("T1", "operation 2 complete");

        data.M3.WaitSort2();                                        // 12

        data.M1.operation3();                                       // 13
        data.printMessage("T1", "operation 3 complete");

        int a1 = data.M1.operation4(0, data.H);                     // 14
        data.printMessage("T1", "operation 4 complete");

        data.M1.calc_a(a1);                                         // 15
        data.printMessage("T1", "operation 5 complete");

        data.M2.SignalA();                                          // 16-18
        data.M2.WaitA();                                            // 19-21

        int[] A1 = data.M1.get_A();                                 // 22
        int[][] MX1 = data.M1.get_MX();                             // 23
        a1 = data.M1.get_a();                                       // 24

        data.M1.operation6(A1, MX1, a1, 0, data.H);                 // 25
        data.printMessage("T1", "operation 6 complete");

        data.M2.SignalOut();                                        // 26
    }
}
